src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -214,6 +194,73 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Split the secret annotation from the delimiter provided and decides whether to use the legacy provider
+     * or a novel provider to obtain the secret value of the requested alias.
+     *
+     * @param secretAnnotation String contains the alias, the provider type and the repository type.
+     * @return Plain text value of the required secret.
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Getting secrets from legacy provider.");
+                }
+                return getSecret(secretAnnotation);
+            }
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Getting secrets from a provider other than the legacy provider.");
+        }
+        return resolveSecret(secretAnnotationStrings);
+    }
+
+    /**
+     * Resolve the secret annotation for the secrets coming from repositories belongs to providers listed under
+     * secretProviders property.
+     * Repositories categorized under providers other than legacy providers will be identified based on the secret
+     * annotation. Under that, When there is only a single repository, provider type and the repository type should
+     * be assigned otherwise, annotation itself is going to be used to get the above-said values.
+     *
+     * @param annotation Value retrieve by the resolveSecret as the value to be resolved.
+     * @return If there is a secret , otherwise , alias itself.
+     */
+    private String resolveSecret(String[] annotation) throws SecureVaultException {
+
+        int length = annotation.length;
+
+        switch (length) {
+            case 1:
+                if (providers.isEmpty()) {
+                    log.error("No provider has been configured. Returning the annotation itself.");
+                    return Arrays.toString(annotation);
+                }
+                if (secretRepositories.isEmpty()) {
+                    log.error("No repository has been configured. Returning the annotation itself.");
+                    return Arrays.toString(annotation);
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Set the values for the provider and the repository type. Returning the value " +
+                            "for secret annotation.");
+                }
+                return getSecret((String) providers.keySet().toArray()[0],
+                        (String) secretRepositories.keySet().toArray()[0],
+                        annotation[0]);
+            case 3:
```

Comment:
Why don't we have a case 2 here ? Am I overseeing anything ?

Reply:
The secret annotation can contain only either one string value or three, anything apart from that will get handled by the default case.